uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/file/CopyPopupView.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.ext.editor.commons.client.file;
+
+import com.github.gwtbootstrap.client.ui.Button;
+import com.github.gwtbootstrap.client.ui.TextBox;
+import com.github.gwtbootstrap.client.ui.constants.ButtonType;
+import com.github.gwtbootstrap.client.ui.constants.IconType;
+import com.google.gwt.event.dom.client.HasClickHandlers;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.HasText;
+import org.uberfire.ext.editor.commons.client.resources.CommonImages;
+import org.uberfire.ext.editor.commons.client.resources.i18n.CommonConstants;
+import org.uberfire.ext.widgets.common.client.common.popups.FormStylePopup;
+import org.uberfire.ext.widgets.common.client.common.popups.footers.GenericModalFooter;
+
+public class CopyPopupView extends FormStylePopup implements CopyPopup.View {
+
+    final private TextBox nameTextBox = new TextBox();
+    final private TextBox checkInCommentTextBox = new TextBox();
+    private final Button copy;
+    private final Button cancel;
+
+    public CopyPopupView() {
+        super( CommonImages.INSTANCE.edit(),
+               CommonConstants.INSTANCE.CopyPopupTitle() );
+        //Make sure it appears on top of other popups
+        getElement().getStyle().setZIndex( Integer.MAX_VALUE );
+
+        nameTextBox.setTitle( CommonConstants.INSTANCE.NewName() );
+        nameTextBox.setWidth( "200px" );
+        addAttribute( CommonConstants.INSTANCE.NewNameColon(),
+                      nameTextBox );
+
+        checkInCommentTextBox.setTitle( CommonConstants.INSTANCE.CheckInComment() );
+        checkInCommentTextBox.setWidth( "200px" );
+        addAttribute( CommonConstants.INSTANCE.CheckInCommentColon(),
+                      checkInCommentTextBox );
+        hide();
+
+        GenericModalFooter footer = new GenericModalFooter();
+        this.copy = footer.addButton( CommonConstants.INSTANCE.CopyPopupCreateACopy(),
+                                      IconType.SAVE,
+                                      ButtonType.PRIMARY );
```

Comment:
IMO MVP is a conceptual pattern not tied to any specific implementation.
What benefit does GWTP bring (something IIRC we looked into and
discounted)? I did not understand their Proxy interface.

Sent on the move
On 4 May 2015 21:09, "Jiří Locker" <notifications@github.com> wrote:

> In
> uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/file/CopyPopupView.java
> <https://github.com/uberfire/uberfire-extensions/pull/32#discussion_r29618284>
> :
>
> > +
> > +        nameTextBox.setTitle( CommonConstants.INSTANCE.NewName() );
> > +        nameTextBox.setWidth( "200px" );
> > +        addAttribute( CommonConstants.INSTANCE.NewNameColon(),
> > +                      nameTextBox );
> > +
> > +        checkInCommentTextBox.setTitle( CommonConstants.INSTANCE.CheckInComment() );
> > +        checkInCommentTextBox.setWidth( "200px" );
> > +        addAttribute( CommonConstants.INSTANCE.CheckInCommentColon(),
> > +                      checkInCommentTextBox );
> > +        hide();
> > +
> > +        GenericModalFooter footer = new GenericModalFooter();
> > +        this.copy = footer.addButton( CommonConstants.INSTANCE.CopyPopupCreateACopy(),
> > +                                      IconType.SAVE,
> > +                                      ButtonType.PRIMARY );
>
> +1 I like the way with presenter interface better, too.
>
> I wonder if Uberfire could re-use MVP interfaces from the GWTP Project
> <http://dev.arcbees.com/gwtp/features/>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uberfire/uberfire-extensions/pull/32/files#r29618284>.
>